### PR TITLE
Improved Unet for ISICs dataset /s4555042

### PR DESCRIPTION
**Pull request by Echozhe**

### CODE REVIEW

**Comment by Echozhe** [comment]
Dear Dr Chandra,

Here is the COMP3710 report pull request.
My name is Zhe Gong. (ID: 45550421)
Task: Problem 4,  Improved Unet on ISICs dataset.
Please kindly refer to the codes and README.

**Comment by SiyuLiu0329** [comment]
- Commit messages should be more specific
- Code valid for Normal Difficulty
- Test set result > 0.8

Other Feedback
- Please follow python file naming convention e.g. `lorem_ipsum_123_abc.py`
- `up4_` does not follow python variable naming convection. I suggest you use `up_4`

**Comment by rexsw** [comment]
Code looks good for normal difficulty

Your pull request should include a description of what is in it and why it should be merged

Comments such as 
   # Dice coefficient metric
   def dice_coefficient(y_true, y_pred, smooth=1.):
Do not add to the readability of the code and should not be included. 

Code in files such as test_script should be in a main function

File: recognition/ZheGong-Unet/model.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
+import tensorflow as tf
```
Comment (shakes76) [inline recognition/ZheGong-Unet/model.py:1]: Code header block should be added for all scripts/source files

**Comment by shakes76** [comment]
## TF Usage
OK
## Good Practice (Design/Commenting)
Good design and comments
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage, Readme OK
Module present
## Commit Log
Meaningful commit messages 
Progressive commits used 
## Documentation
Description and Comments good
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Proper Request Description. In the future, a more detailed pull request could've been done. When developer receiving the request is not known, the request could be rejected. Remember you could use/reuse the README info.

**Pull request merged**